tests/compile/fsm-guard.fil

```diff
@@ -0,0 +1,54 @@
+// Tests the guards of an fsm.
+
+comp test2[#DELAY]<G: #DELAY>(
```

Comment:
ALL CAPS parameters look pretty ugly but we haven't really discussed any good conventions around naming schemes. Maybe we should do that in a new doc page to ensure consistency and readable code for everyone.